### PR TITLE
Fix DelegatingServiceInstanceListSupplier must

### PR DESCRIPTION
**Pull request by HJK181**
respect `SelectedInstanceCallback` on its delegate. Motivation behind is that some load balancer implementations like the `RoundRobinLoadBalancer` to call `selectedServiceInstance` on the `ServiceInstanceListSupplier` after choosing the next instance. However, when a `ServiceInstanceListSupplier` in wrapped by the `DelegatingServiceInstanceListSupplier` this functionality breaks, as `DelegatingServiceInstanceListSupplier` is not implementing `SelectedInstanceCallback` and therefore does ignoring that its delegate might also do. This might break the functionality of the delegate. Fixes gh-1221.

### CODE REVIEW

**Comment by HJK181** [comment]
@spencergibb @OlgaMaciaszek anybody?

**Comment by OlgaMaciaszek** [comment]
Thanks for providing the PR, @HJK181. Will take a look at it.

File: spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/TestSelectedServiceInstanceSupplier.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
```
Comment (OlgaMaciaszek) [inline spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/TestSelectedServiceInstanceSupplier.java:2]: Please change to `* Copyright 2012-2023 the original author or authors.`

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Hello @HJK181, thanks for submitting the PR. It looks good, but there are some cosmetic changes that need to be made - please address the comments. Also, please update the forst line of the copyright clause on top of each changed file to `-2023`.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DelegatingServiceInstanceListSupplier.java [hunk 2]
```diff
@@ -26,9 +27,10 @@
  *
  * @author Spencer Gibb
  * @author Olga Maciaszek-Sharma
+ * @author hjk181
```
[the remark this answers is not in the document]
Comment (HJK181) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DelegatingServiceInstanceListSupplier.java]: done

File: spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/TestSelectedServiceInstanceSupplier.java (hunk 1 above)
Comment (HJK181) [reply spring-cloud-loadbalancer/src/test/java/org/springframework/cloud/loadbalancer/core/TestSelectedServiceInstanceSupplier.java]: done

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
Thanks, @HJK181 . Have added one more comment. Please take a look.

File: spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DelegatingServiceInstanceListSupplier.java [hunk 3]
```diff
@@ -26,9 +27,10 @@
  *
  * @author Spencer Gibb
  * @author Olga Maciaszek-Sharma
+ * @author Jürgen Kreitler
```
[the remark this answers is not in the document]
Comment (HJK181) [reply spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/DelegatingServiceInstanceListSupplier.java:30]: Done with force push of the last commit.

**Review by OlgaMaciaszek — APPROVED** [review]
Thanks @HJK181. LGTM.

**Pull request merged**